.github/workflows/ci-tests.yml

```
@@ -47,37 +47,34 @@ jobs:
           go-version-file: go.mod
           cache-dependency-path: go.sum
 
-      - name: Cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cache/golangci-lint
-          key: 'golangci-lint-${{ runner.os }}-${{ hashFiles(''**/go.sum'') }}'
-          restore-keys: |
-            golangci-lint-${{ runner.os }}-
-
       - name: Setup CI Tooling
         uses: shrink/actions-docker-extract@v3
         with:
           image: tykio/ci-tools:latest
           path: /usr/local/bin/golangci-lint
           destination: /usr/local/bin
 
-      - run: golangci-lint version && golangci-lint cache status
-
-      - name: golangci-lint
-        if: ${{ github.event_name == 'pull_request' }}
-        # This is a pull request, but exiting here skips CI tests (`needs: golangci-lint`). Exit code 0 to always run tests.
-        # PR uses the "prod" config and must pass.
+      - name: Run codegen and tidy
+        if: github.event_name == 'pull_request'
+        env:
+          BASE_BRANCH: ${{ github.base_ref || 'master' }}
         run: |
-          golangci-lint run --out-format colored-line-number,checkstyle:golangci-lint-report.json \
-                            --issues-exit-code=0 --new-from-rev=origin/${{ env.BRANCH_NAME }} -v ./...
+          # Run linting without auto-fixing
+          task tidy
+          task codegen:smart
```

`codegen:smart` is duplicated in `tidy` step
```suggestion
          task tidy
```